storage/innobase/log/log0recv.cc

```diff
@@ -3341,6 +3341,7 @@ static buf_block_t *recv_recover_page(buf_block_t *block, mtr_t &mtr,
 			}
 
 			mtr.discard_modifications();
+			block->page.lock.x_lock();
 			mtr.commit();
```

Comment:
Where is the page latch going to be released? Why does this not trigger a failure of `buf_pool_t::assert_all_freed()` on shutdown? Note: this may conflict with #4213 as well. If this change is truly necessary, then I think that it deserves a source code comment.

If this is fixing the crash in crash recovery that I noted in [MDEV-37192](https://jira.mariadb.org/browse/MDEV-37192), I think that it needs to be filed and fixed separately, with a test case. Does this bug affect 10.6 as well?